### DISCUSSION TITLE
Unable to click on links within the Property ToolTip

### DISCUSSION THREAD

**Discussion by CodingOctocat** [godotengine/godot-proposals#7679] [Editor]
Tooltips for properties should be delayed to disappear, Otherwise I'd never be able to click through to the link inside.

![动画](https://github.com/godotengine/godot-proposals/assets/7220248/ec915ad8-7d9e-4920-96ba-8ab1d8ff82de)

**Comment by aXu-AP** [comment]
Yes, this is frustrating. In practice, this needs 2 steps:

1. There should be a short period of time when you move the mouse, you can move it on top of the tooltip and it shouldn't disappear.
2. Allow links to be pressed, opening the docs.

**Comment by dalexeev** [comment]
Note that you can use context menu to open the documentation.

![](https://github.com/godotengine/godot-proposals/assets/47700418/ba3dba6c-c2ad-44c7-99ce-e93d11fe8e4f)

**Comment by azur-wolve** [comment]
have the ability to freeze those tooltips by holding `Alt` could be the thing

**Reply by Calinou** [reply]
This doesn't sound bad in principle, but it sounds difficult for users to discover this feature. To my knowledge, this isn't standard UI behavior on any OS.

**Reply by CodingOctocat** [reply]
Delayed closure is standard behavior, and I remember menus on web pages starting to use this as early as years ago.

**Reply by Calinou** [reply]
The issue with delayed closure is that it would make editing in the inspector feel a lot less snappy, since you'd need to wait before you can see a property that is behind a tooltip that would linger on for a while after moving the mouse.

Tooltips can be pretty large, so we need to make sure they don't stay visible for longer than they need to when you want to get them out of your way.

**Reply by CodingOctocat** [reply]
@Calinou What about MS Office's toolbar behavior?
![动画](https://github.com/godotengine/godot-proposals/assets/7220248/be60ea33-411d-46b6-b64f-297f9ededce7)

**Reply by Calinou** [reply]
Since tooltips are separate windows, they can't always be semi-transparent on all platforms (namely, on Linux with compositing disabled in the window manager, or [if per-pixel transparency fails to kick in due to system configuration](https://github.com/godotengine/godot/issues/76167) or engine bugs).

**Reply by CodingOctocat** [reply]
Perhaps the translucency feature is not needed and only distance-based closing behavior is used.

**Reply by aXu-AP** [reply]
I think this wouldn't be too bad. If cursor is moving away from the tooltip, hide. If cursor is moving towards the tooltip, wait a small time to see if cursor ends on top of it.
Hiding should be much faster than in MS office, though. That context toolbar is way too persistent (and for a tooltip even moreso).

**Comment by jsjtxietian** [comment]
In this issue (https://github.com/godotengine/godot/issues/86110) I wrote some my naive thoughts, there are two behaviours : "delay to disappear" and "stay on hover". In my naive thought, if give a pretty small delayed closure time maybe it's enough for use to move mouse on the tooltip but will not feel snappy? This needs some test.
